nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2019 Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__BT_SERVICE_NODE_HPP_
+#define NAV2_BEHAVIOR_TREE__BT_SERVICE_NODE_HPP_
+
+#include <string>
+#include <memory>
+
+#include "behaviortree_cpp/action_node.h"
+#include "nav2_util/node_utils.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_behavior_tree
+{
+
+template<class ServiceT>
+class BtServiceNode : public BT::CoroActionNode
```

Comment:
Ah, I just took the naming from `BtActionNode` since it implemented a rclcpp action, and this is a `BtServiceNode` with a service client. Looks like that Action referred to BT not rclcpp.

Is there another name you prefer? I'm not married to anything